### PR TITLE
Ahmad

### PR DESCRIPTION
**Pull request by ahmadrv**
@Requestparam --> @RequestEntity

### CODE REVIEW

File: src/main/java/ir/ui/golestan/data/repository/ScoreRepository.java [hunk 1]
```diff
@@ -13,6 +13,5 @@
 public interface ScoreRepository extends JpaRepository<Score, Score.ScoreId> {
     @Query("SELECT s FROM Score s WHERE s.id = :id")
     Collection<Score> findScoresById(@Param("id") int id);
```
Comment (sbaneshi) [inline src/main/java/ir/ui/golestan/data/repository/ScoreRepository.java]: 1. in method ke kolan estefade nashode. 
2. ham query haye sade ro mishe ba esm method moshakhas kard. spring khodesh mofahme. to intellij `find` ro benevisi baraye esm method khodesh ye list baz mikone komaket mikone
3. method haye sade ro JpaRepository dare to khodesh.mesle finalAll, findById, ...

File: src/main/java/ir/ui/golestan/restservice/ProfessorController.java [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
 
 	@GetMapping("/professor")
-    public void addScore(@RequestParam(value = "studentId") int studentId,
-                         @RequestParam(value = "courseId") int courseId,
-                         @RequestParam(value = "score") double score,
-                         ScoreRepository repository) {        // Repository???
-
-        System.out.println(String.format(template, studentId, courseId, score));
+    public void setStudentScore(RequestEntity<?> request, int studentId, int courseId, 
+                                double score, ScoreRepository repository) {// Repository???
```
Comment (sbaneshi) [inline src/main/java/ir/ui/golestan/restservice/ProfessorController.java]: RestController va har class dige ee ke ba @Component va farzandanesh alamat gozari beshe to spring mishe ye `bean` .
yani spring miad nega be constructor in class mikone va har chi ke lazaem bashe ro behesh mide va ye shei azah dorost mikone va to khodesh negah midare. va baad har jaye dige ye class dige in ro khast behesh mide. alan ma ScoreRepositiry to in class lazem darim. mizarimesh to counstructoresh. zemn in ke controler ha bayad az BaseController ers bebarand ta method getAuthenticatedUser ro betonim estefade konim

File: src/main/java/ir/ui/golestan/restservice/ProfessorController.java [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
+        
+        AuthenticatedUser user = getAuthenticatedUser(request, Role.PROFESSOR);
         repository.save(new Score(studentId, courseId, score)); // @AllArgsConstructor?
```
Comment (sbaneshi) [inline src/main/java/ir/ui/golestan/restservice/ProfessorController.java]: az Score.builder() baraye sakht ashya estefade konim behtare. builder() az gozashtan @Builder balaye class Score miad

Comment (sbaneshi) [reply src/main/java/ir/ui/golestan/restservice/ProfessorController.java]: Score.builder()
	.studentId(studentId)
	.courseId(courseId)
	.score(score)
	.build()

File: src/main/java/ir/ui/golestan/restservice/ProfessorController.java [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
 import ir.ui.golestan.data.entity.Score;
 import ir.ui.golestan.data.repository.ScoreRepository;
+import ir.ui.golestan.authorization.AuthenticatedUser;
+import ir.ui.golestan.authorization.Role;
 
 @RestController
 public class ProfessorController {
-    private static final String template = "***User ID: %d, Score: %d, Course ID: %. 2f***";
 
 	@GetMapping("/professor")
```
Comment (sbaneshi) [inline src/main/java/ir/ui/golestan/restservice/ProfessorController.java]: /professor/set_score

File: src/main/java/ir/ui/golestan/data/repository/ScoreRepository.java (hunk 1 above)
Comment (ahmadmahmoodiandarvishani) [inline src/main/java/ir/ui/golestan/data/repository/ScoreRepository.java:15]:
```suggestion
    Collection<Score> findScoresBystudentId(int studentId);
```
intellij nadarm. VScode daram mizanam. Alan injuri okye?
badam tu StudentController seda mizanam ba @requestEntity o auth.
@sbaneshi

File: src/main/java/ir/ui/golestan/restservice/ProfessorController.java (hunk 4 above)
Comment (ahmadmahmoodiandarvishani) [inline src/main/java/ir/ui/golestan/restservice/ProfessorController.java:17]:
```suggestion
	@GetMapping("/professor/set_score")
```

File: src/main/java/ir/ui/golestan/data/repository/ScoreRepository.java (hunk 1 above)
Comment (ahmadmahmoodiandarvishani) [reply src/main/java/ir/ui/golestan/data/repository/ScoreRepository.java:15]:
```suggestion
    Collection<Score> findScoresBystudentId(int studentId);
```
intellij nadarm. ba VSCode darm mizanm.
in okye?
badam tu student controller sedash mizanm ba @requestEntity o auth.

File: src/main/java/ir/ui/golestan/restservice/ProfessorController.java (hunk 4 above)
Comment (ahmadrv) [reply src/main/java/ir/ui/golestan/restservice/ProfessorController.java]:
```suggestion
	@GetMapping("/professor/set_score")
```

File: src/main/java/ir/ui/golestan/restservice/ProfessorController.java (hunk 2 above)
Comment (ahmadrv) [reply src/main/java/ir/ui/golestan/restservice/ProfessorController.java]:
```suggestion
    public class ProfessorController extends BaseController {

    private final ScoreRepository scoreRepository;

    public ProfessorController(GolestanConfiguration configuration, AuthorizationService authorizationService,
            ScoreRepository scoreRepository) {
        super(configuration, authorizationService);
        this.scoreRepository = scoreRepository;
    }
```
in Okye?

File: src/main/java/ir/ui/golestan/restservice/ProfessorController.java (hunk 3 above)
Comment (ahmadrv) [reply src/main/java/ir/ui/golestan/restservice/ProfessorController.java]:
```suggestion
        repository.save(Score.builder().studentId(studentId).courseId(courseId).score(score).build());
```

File: src/main/java/ir/ui/golestan/data/repository/ScoreRepository.java (hunk 1 above)
Comment (ahmadrv) [reply src/main/java/ir/ui/golestan/data/repository/ScoreRepository.java]:
```suggestion
    List<Score> findByStudentId(int studentId);
```
in okye?
https://stackoverflow.com/a/27627297

to StudentController hm sedash mikonm ba @requestEntity o auth.

Comment (sbaneshi) [reply src/main/java/ir/ui/golestan/data/repository/ScoreRepository.java]: auth ba UserRole kar mikone. 
baraye bargardondan chandta bayad findAll bezani -> findAllByStudentId
gofftam ke. ti intellij to hmin interface benevis 'find' ye lest barat baz mishe baraye komak. 

method StudentController ham faqat ye RequestEntity niaz dare. sutudentId ro ba auth mishe dar avord

File: src/main/java/ir/ui/golestan/restservice/ProfessorController.java (hunk 2 above)
Comment (sbaneshi) [reply src/main/java/ir/ui/golestan/restservice/ProfessorController.java]: are

File: src/main/java/ir/ui/golestan/data/repository/ScoreRepository.java (hunk 1 above)
Comment (ahmadrv) [reply src/main/java/ir/ui/golestan/data/repository/ScoreRepository.java]: intellij nadarm. ba VSCode darm mizanm.

File: src/main/java/ir/ui/golestan/data/repository/ScoreRepository.java [hunk 5]
```diff
@@ -2,17 +2,14 @@
...
-import org.springframework.data.jpa.repository.Query;
-import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface ScoreRepository extends JpaRepository<Score, Score.ScoreId> {
-    @Query("SELECT s FROM Score s WHERE s.id = :id")
-    Collection<Score> findScoresById(@Param("id") int id);
 
+    List<Score> findScoresBystudentId(int studentId);
```
Comment (ahmadrv) [inline src/main/java/ir/ui/golestan/data/repository/ScoreRepository.java]:
```suggestion
    List<Score> findAllByStudentId(int studentId);
```

**Review by ahmadrv — COMMENTED** [review]
ye nega b changed files hm bendaz bbin oonam okye?

**Pull request merged**